.github/workflows/ci.yml

```diff
@@ -28,7 +33,7 @@ jobs:
           BLACK_VERSION="$(python tools/dev/lint_version.py black poetry.lock)"
           pip install ruff=="$RUFF_VERSION" black=="$BLACK_VERSION"
       - name: Set up Terraform
-        uses: hashicorp/setup-terraform@v3
+        uses: hashicorp/setup-terraform@b9cd54a3c349d3f38e8881555d616ced269862dd # v3.1.2
```

Comment:
any pinning here requires a corresponding periodic update, ideally with dependabot group, afaics that's missing in this pr.

Reply:
I believe dependabot will update these hashes when it runs automatically, as long as github actions is set to be updated in the dependablt yml file

Reply:
yeah. i just saw we already had dependabot group configuration for action updates when making the other pr.